File: fightnight.Server/Models/Event.cs
using fightnight.Server.Enums;
using fightnight.Server.Models;

namespace fightnight.Server.models
{
    public class Event
    {
        //normalise by taking venue and organiser 

        public string id { get; set; } = Guid.NewGuid().ToString();
        public string title { get; set; } = string.Empty;
        public DateTime dateTime { get; set; }
        public TimeSpan eventDur { get; set; }
        public string venue { get; set; } = string.Empty;
        public string VenueAddress { get; set; } = string.Empty;
        public string desc {get; set;} = string.Empty;
        public string type { get; set; } = string.Empty;
        public EventStatus status { get; set; } = EventStatus.Planning;
        public string organizer { get; set; } = string.Empty;
        public string adminId { get; set; } = string.Empty;
        public int numRounds { get; set; } = 3;
        public int roundDur { get; set; } = 3;
        //participant list
        //mod list
        //img

        public List<AppUserEvent> AppUserEvents { get; set; } = new List<AppUserEvent>();

    }
}

File: fightnight.client/src/Models/Event.ts
export type Event = {
    id: string,
    title: string,
    date: Date,
    time: string,
    eventDur: string,
    venue: string,
    venueAddress: string,
    desc: string,
    type: string,
    status: number,//EventStatus
    organizer: string,
    adminId: string,
    numMatches: number,
    numRounds: number,
    roundDur: number,
    role: number,//EventRole, 
    bannerUrl: string
}

export enum EventStatus {
    Planning,
    Launched,
    Ongoing,
    Completed
}

export enum EventRole {
    Spectator,
    Moderator,
    Fighter,
    Admin
}